baremetal/metalmachinetemplate_manager.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package baremetal
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	capm3 "github.com/metal3-io/cluster-api-provider-metal3/api/v1alpha4"
+	"github.com/pkg/errors"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TemplateManagerInterface is an interface for a TemplateManager
+type TemplateManagerInterface interface {
+	UpdateAutomatedCleaningMode(context.Context) error
+}
+
+// MachineTemplateManager is responsible for performing metal3MachineTemplate reconciliation
+type MachineTemplateManager struct {
+	client client.Client
+
+	Metal3MachineList     *capm3.Metal3MachineList
+	Metal3MachineTemplate *capm3.Metal3MachineTemplate
+	Log                   logr.Logger
+}
+
+// NewMachineTemplateManager returns a new helper for managing a metal3MachineTemplate
+func NewMachineTemplateManager(client client.Client,
+	metal3MachineTemplate *capm3.Metal3MachineTemplate,
+
+	metal3MachineList *capm3.Metal3MachineList,
+	metal3MachineTemplateLog logr.Logger) (*MachineTemplateManager, error) {
+
+	return &MachineTemplateManager{
+		client: client,
+
+		Metal3MachineTemplate: metal3MachineTemplate,
+		Metal3MachineList:     metal3MachineList,
+		Log:                   metal3MachineTemplateLog,
+	}, nil
+}
+
+// UpdateAutomatedCleaningMode iterates through metal3Machines and returns a list of metal3machines referencing
+// a metal3MachineTemplate object that they are cloned from.
```

Comment:
Is it only listing or also syncing the `AutomatedCleaningMode` ?

Reply:
Good catch, fixing it